src/ts_catalog/continuous_agg.c

```diff
@@ -77,6 +90,52 @@ ts_continuous_agg_with_clause_parse(const List *defelems)
 								 continuous_aggregate_with_clause_def,
 								 TS_ARRAY_LEN(continuous_aggregate_with_clause_def));
 }
+
+List *
+ts_continuous_agg_unparse_compression_defelems(const WithClauseResult *with_clauses)
+{
+	List *ret = NIL;
+
+	for (int i = 0; i < _CompressOptionMax; i++)
+	{
+		int option_index = 0;
+		switch (i)
+		{
+			case CompressEnabled:
+				option_index = ContinuousViewOptionCompress;
+				break;
+			case CompressSegmentBy:
+				option_index = ContinuousViewOptionCompressSegmentBy;
+				break;
+			case CompressOrderBy:
+				option_index = ContinuousViewOptionCompressOrderBy;
+				break;
+			case CompressChunkTimeInterval:
+				option_index = ContinuousViewOptionCompressChunkTimeInterval;
+				break;
+			default:
+				ereport(ERROR,
```

Comment:
For internal program errors, there's a shorter form `elog(ERROR, ...` which also sets the right error code for you. Or `Ensure` if you want it to be an assertion in debug.

Reply:
thank you; I was using ereport because it seemed `elog` was deprecated...but its more convinient for sure! :)